src/Api/ExceptionHandler/FallbackExceptionHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Api\ExceptionHandler;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use Tobscure\JsonApi\Exception\Handler\ExceptionHandlerInterface;
+use Tobscure\JsonApi\Exception\Handler\ResponseBag;
+
+class FallbackExceptionHandler implements ExceptionHandlerInterface
+{
+    /**
+     * @var bool
+     */
+    protected $debug;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param bool $debug
+     * @param LoggerInterface $logger
+     */
+    public function __construct($debug, LoggerInterface $logger)
+    {
+        $this->debug = $debug;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function manages(Exception $e)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Exception $e)
+    {
+        $status = 500;
+        $error = $this->constructError($e, $status);
+
+        if (! $this->debug) {
+            $this->logger->error($e);
+        }
```

Comment:
Why only log in debug mode?

Reply:
My thought was that in debug mode you can see the error displayed when you encounter it, and therefore no need to log it. I suppose it'd be simpler to just log everything regardless though.

Reply:
On second thought, errors that occur in the forum stack don't get logged when debug mode is on, because they go through the HandleErrorsWithWhoops middleware. Whatever we do, it should be consistent between the forum/API stacks. Should we add logging to the Whoops middleware or keep things as-is?

Reply:
Yes, I agree. Always logging means a) no surprises and b) persistence.